client.go

```diff
@@ -96,7 +97,13 @@ func HijackHttpRequest(options HijackHttpOptions) error {
 	clientconn := httputil.NewClientConn(dial, nil)
 	defer clientconn.Close()
 
-	clientconn.Do(req)
+	res, err := clientconn.Do(req)
+	if res.StatusCode >= http.StatusBadRequest {
+		if err := options.ErrorHandler(res); err != nil {
```

Comment:
doesn't this code hide the original error? We at least should pass it into the ErrorHandler to do sthg with it.